spec/rspec/rails/matchers/have_enqueued_mail_spec.rb

```diff
@@ -95,6 +99,10 @@ def test_email; end
       expect { }.not_to have_enqueued_email
     end
 
+    it "passes when negated with 0 arguments and a non-mailer job is enqueued" do
+      expect { NonMailerJob.perform_later }.not_to have_enqueued_email
```

Comment:
(This new spec fails without the `job_match?` => `job_matches?` rename done in this PR.)